packages/fluentui/react-northstar/src/utils/positioner/usePopper.ts

```diff
@@ -0,0 +1,391 @@
+ * in React DOM and will not pass "autoFocus" to an actual HTML.
+ *
+ * @param {Node} node
+ * @returns {Boolean}
+ */
+function hasAutofocusProp(node: Node): boolean {
+  // https://github.com/facebook/react/blob/848bb2426e44606e0a55dfe44c7b3ece33772485/packages/react-dom/src/client/ReactDOMHostConfig.js#L157-L166
+  const isAutoFocusableElement =
+    node.nodeName === 'BUTTON' ||
+    node.nodeName === 'INPUT' ||
+    node.nodeName === 'SELECT' ||
+    node.nodeName === 'TEXTAREA';
+
+  if (isAutoFocusableElement) {
+    return !!getReactFiberFromNode(node).pendingProps.autoFocus;
+  }
+
+  return false;
+}
+
+function hasAutofocusFilter(node: Node) {
+  return hasAutofocusProp(node) ? NodeFilter.FILTER_ACCEPT : NodeFilter.FILTER_SKIP;
+}
+
+/**
+ * Provides a callback to resolve Popper options, it's stable and should be used as a dependency to trigger updates
+ * of Popper options.
+ *
+ * A callback is used there intentionally as some of Popper.js modifiers require DOM nodes (targer, container, arrow)
+ * that can't be resolved properly during an initial rendering.
+ *
+ * @param {UsePopperOptions} options
+ * @param {React.MutableRefObject} popperOriginalPositionRef
+ *
+ * @returns {Function}
+ */
+function usePopperOptions(options: UsePopperOptions, popperOriginalPositionRef: React.MutableRefObject<string>) {
+  const {
+    autoSize,
+    flipBoundary,
+    offset,
+    onStateUpdate,
+    overflowBoundary,
+    rtl,
+    unstable_disableTether,
+    unstable_pinned,
+  } = options;
+
+  const placement = getPlacement(options.align, options.position, options.rtl);
+  const strategy = options.positionFixed ? 'fixed' : 'absolute';
+
+  const handleStateUpdate = useEventCallback(({ state }: { state: Partial<PopperJs.State> }) => {
+    if (onStateUpdate) {
+      onStateUpdate(state);
+    }
+  });
+
+  const offsetModifier = React.useMemo(
+    () =>
+      offset
+        ? {
+            name: 'offset',
+            options: { offset: rtl ? applyRtlToOffset(offset) : offset },
+          }
+        : null,
+    [offset, rtl],
+  );
+  const userModifiers = useDeepMemo(() => options.modifiers, options.modifiers);
+
+  return React.useCallback(
+    (
+      target: HTMLElement | PopperJs.VirtualElement,
+      container: HTMLElement,
+      arrow: HTMLElement | null,
+    ): PopperJs.Options => {
+      const scrollParentElement: HTMLElement = getScrollParent(container);
+      const hasScrollableElement = scrollParentElement
+        ? scrollParentElement !== scrollParentElement.ownerDocument.body
+        : false;
+
+      const modifiers: PopperJs.Options['modifiers'] = [
+        /**
+         * We are setting the position to `fixed` in the first effect to prevent scroll jumps in case of the content
+         * with managed focus. Modifier sets the position to `fixed` before all other modifier effects. Another part of
+         * a patch modifies ".forceUpdate()" directly after a Popper will be created.
+         */
+        {
```

Comment:
This modifier has been moved up in the array. Could you please explain?

Reply:
https://github.com/microsoft/fluentui/pull/17276#discussion_r589292053